### PR TITLE
Improved lazy loading of prettier by only requiring it when it's first used

### PR DESCRIPTION
**Pull request by mattpocock**
This PR ensures that prettier is now only loaded when it's actually used for formatting for the first time. It turns `prettier.format` into an async call, which the first time it's called loads prettier via CDN (with a 15 second timeout).

Since it's loaded via CDN, subsequent requests to the page are done to the disk cache.

If you like this, we could also do this with Elk.

### CODE REVIEW

**Comment by davidkpiano** [comment]
> If you like this, we could also do this with Elk.

Yes please!

**Review by davidkpiano — APPROVED** [review]

File: src/prettier.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  export const prettier: typeof import('prettier');
+  export const prettierPlugins: (string | import('prettier').Plugin)[];
+
+  interface Window {
+    define: any;
+  }
+}
+
+const Prettier = () => {
+  let hasLoaded = false;
```
Comment (davidkpiano) [inline src/prettier.ts:28]: Implicit state machine 🙈  (fine for now)

File: src/prettier.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     * Helps Monaco editor get over a too-sensitive error
+     *
+     * https://stackoverflow.com/questions/55057425/can-only-have-one-anonymous-define-call-per-script-file
+     */
+    const define = window.define;
+    window.define = () => {};
+
+    await Promise.all([
+      loadScript('https://unpkg.com/prettier@2.3.2/standalone.js'),
+      loadScript('https://unpkg.com/prettier@2.3.2/parser-typescript.js'),
```
Comment (farskid) [inline src/prettier.ts:43]:
```suggestion
      loadScript('https://unpkg.com/prettier@latest/standalone.js'),
      loadScript('https://unpkg.com/prettier@latest/parser-typescript.js'),
```

**Review by farskid — APPROVED** [review]

File: src/prettier.ts (hunk 2 above)
Comment (mattpocock) [reply src/prettier.ts:43]: Is this a good idea? I think it might be better to fix it to a specific version

File: src/prettier.ts (hunk 1 above)
Comment (mattpocock) [reply src/prettier.ts:28]: 😂  I figured this was easier than trying to wrangle `service.send` inside an async function

**Comment by davidkpiano** [comment]
Is this good to go in?

**Pull request merged**